src/doi_tools.py

```diff
@@ -18,9 +18,10 @@ def extractDOI(inputstring):
 
     return result
 
-def extractURLFromString(inputstring):
+def extract_url_from_string(inputstring):
     try:
         result = re.search("(?P<url>https?://[^\s]+)", inputstring).group("url").replace('\"', '')
     except:
         result = "NOT FOUND"
+        raise
```

Comment:
We don't want to do a `raise` here since this is an exception that we accept. We can address this [code smell](https://sonarcloud.io/project/issues?id=usdot-its-jpo-data-portal_datahub-ingest&open=AXN7QqSfSVCEXV1-KT5R&resolved=false&severities=CRITICAL) by adding a more specific exception though. Something like "UrlNotExtracted" Error could work.

Some documentation on how to create user-defined exceptions [here](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions).